nni/runtime/tuner_command_channel/channel.py

```diff
@@ -48,6 +56,35 @@ def disconnect(self) -> None:
     #     ...
     # def receive(self) -> Command | None:
     #     ...
+    def send(self, command: Command) -> None:
+        command_dict = command.__dict__
+        command_json = json.dumps(command_dict)
+        self._channel.send(command_json)
+
+    def receive(self) -> Command | None:
+        command_json = self._channel.receive()
+        if command_json is None:
+            raise RuntimeError('NNI manager closed connection')
+        command = json.loads(command_json)
+        if command['command_type'] == 'ReportMetricData':
```

Comment:
There can be a factory function to avoid if-else.
https://github.com/microsoft/nni/blob/master/nni/experiment/config/utils/internal.py#L171